### ISSUE TITLE
Style of command is applied to previous line

### ISSUE THREAD

**Issue by PatrickRogg** [outline/rich-markdown-editor#233]
Hello,

I used the example provided in the repository and noticed that the style is applied to the previous line when a command is executed in the last line of the editor. This bug occurs in the example of this repository and I also when I integrate the editor in my own application. In Outline itself this error does not occur.

**To Reproduce**
1. Start the example application
2. Add text to the first line
3. Press enter to go to the next line
4. Use a command like '/h1' and press enter.
5. Style will be applied to the text above

**Expected behavior**
The style should be applied to the current line and not the previous line.

**Version**
10.1.2

**Video**
Here is a short video of the problem: [click](https://streamable.com/s8qjdm)

**Desktop (please complete the following information):**
 - OS: Ubuntu 19.04
 - Browser: Chrome (Version: 83), Firefox (Version: 78) (I did not test other browsers, but I would assume that there the same problems occurs)

**Comment by tommoor** [comment]
I'm unable to reproduce this on the latest version, there have been several fixes since `10.1.2` – can you give it a try?

**Comment by PatrickRogg** [comment]
I tried using a newer version, but the problem is still there. Here is a link to the repo: [click](https://github.com/PatrickRogg/rich-markdown-editor-test). It's basically just the default react app with the editor.

**Comment by VarunRustomji** [comment]
I am also noticing this issue. An important thing to note is that this only happens on the last line when there is text on the penultimate line. It doesn't seem to happen when the penultimate line is empty.

**Comment by tommoor** [comment]
Although this has been reported twice I'm still unable to reproduce with the steps given 😢

**Comment by VarunRustomji** [comment]
Can you clone the repo from [here](https://github.com/VarunRustomji/editor-test) and see if you observe the behaviour I mentioned? It is weird because I cloned the example, stripped down the code to bare minimum and it worked fine. So even I'm not sure what is going on.

**Comment by anojht** [comment]
I also compiled the example repo from @VarunRustomji and I was able to reproduce this behaviour. Here is a gif I made showcasing the behaviour: [https://imgur.com/a/Pvt0l9t](https://imgur.com/a/Pvt0l9t)

**Comment by alanqchen** [comment]
I'm having the opposite issue but still similar, the style is being applied to the **next** line ([gif](https://i.imgur.com/YGp2xYS.gif)).

The code used for the test shown is:
```js
import Editor from 'rich-markdown-editor';

const Index = props => (
   <Editor />
);

export default Index;
```

A few observations:
- I use Next.js, but the issue is still there even when the component is rendered client-side
- This issue still persists in 10.6.1

**Comment by tommoor** [comment]
So bizarre, thanks for the gif - this is totally reproducible?

**Comment by alanqchen** [comment]
Yes, I've been able to also reproduce the issue using @VarunRustomji's example repo and using the `Editor` component in `create-react-app`.

**Comment by Kukkimonsuta** [comment]
I have same issue as well: https://codesandbox.io/s/async-bush-1kxzj?file=/src/App.js

OS: Windows 10 2004
Browser 1: Edge 84.0.522.59
Browser 2: Firefox 79.0

Note that https://www.getoutline.com/ doesn't show this behavior.

**Comment by Kukkimonsuta** [comment]
@tommoor I've been able to reproduce it using sample in this repo as well:

1. download clean repo
2. run `yarn install`
3. run `yarn start`
4. go to http://localhost:8080 and it works correctly
5. stop watcher
6. run `yarn upgrade`
7. run `yarn start`
8. go to http://localhost:8080 and an issue occurs

**Comment by tommoor** [comment]
Well, that explains it at least – by running `yarn upgrade` you're installing a different set of dependency versions. I'd recommend using:

`yarn install --pure-lockfile` 

So it seems like the incompatibility is with an upgraded dependency version.

**Comment by tommoor** [comment]
Fixed in `v10.6.2`
